test/e2e/network/service.go

```diff
@@ -2947,6 +2948,118 @@ var _ = SIGDescribe("Services", func() {
+		framework.ExpectNoError(err, "failed to create service")
+
+		svcWatchChan := svcWatch.ResultChan()
+		for event := range svcWatchChan {
+			if event.Type == watch.Added {
+				break
+			}
+		}
+
+		ginkgo.By("patching the ServiceStatus")
+		serviceStatusPatch, err := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]string{"test-service": "patched"},
+			},
+			"spec": map[string]interface{}{
+				"ports": []map[string]interface{}{{
+					"name":       "http8080",
+					"port":       int32(8080),
+					"targetPort": int(8080),
+					"selector": []map[string]interface{}{{
+						"type": "LoadBalancer",
+					}},
+				}},
+			},
+		})
+		framework.ExpectNoError(err, "Could not marshal json", err)
+		_, err = dc.Resource(svcResource).Namespace(ns).Patch(context.TODO(), testServiceName, types.StrategicMergePatchType, []byte(serviceStatusPatch), metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err, "Could not patch service status", err)
+
+		for event := range svcWatchChan {
+			if event.Type == watch.Modified {
+				break
+			}
+		}
+
+		svcStatus, err := dc.Resource(svcResource).Namespace(ns).Get(context.TODO(), testServiceName, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err, "Could not get service status", err)
+
+		var svcStatusGet v1.Service
+		svcStatusUjson, err := json.Marshal(svcStatus)
+		framework.ExpectNoError(err, "Failed to marshal json of replicationcontroller label patch", err)
+
+		json.Unmarshal(svcStatusUjson, &svcStatusGet)
+		if !(svcStatusGet.ObjectMeta.Labels["test-service"] == "patched") {
+			framework.Logf("failed to patch the Service")
+		}
+
+		ginkgo.By("updating the ServiceStatus")
+		svcStatusGet.Spec.Ports[0].Name = "http8081"
+		svcStatusGet.Spec.Ports[0].Port = int32(8081)
+		svcStatusGet.ObjectMeta.Labels["test-service"] = "updated"
+		_, err = cs.CoreV1().Services(ns).Update(context.TODO(), &svcStatusGet, metav1.UpdateOptions{})
+		framework.ExpectNoError(err, "Unable to update service. %v", err)
+
+		for event := range svcWatchChan {
+			if event.Type == watch.Modified {
+				break
+			}
+		}
+
+		ginkgo.By("finding Service in list")
+		svcs, err := cs.CoreV1().Services("").List(context.TODO(), metav1.ListOptions{LabelSelector: "test-service-static=true"})
+		framework.ExpectNoError(err, "Could not list services", err)
+
+		foundSvc := false
+		for _, svcItem := range svcs.Items {
+			if svcItem.ObjectMeta.Name == testServiceName &&
+				svcItem.ObjectMeta.Namespace == ns &&
+				svcItem.ObjectMeta.Labels["test-service"] == "updated" &&
+				svcItem.Spec.Ports[0].Name == "http8081" &&
+				svcItem.Spec.Ports[0].Port == int32(8081) {
+				foundSvc = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundSvc, true, "unable to find Service in list of Services.")
+
+		ginkgo.By("deleting the service")
```

Comment:
suggest move this  after create service(line 2976). use `defer` to confirm service will be removed whether processe failed after service  
created.

Reply:
After the test runs the e2e framework cleans up any resources especially those in the test namespace

Reply:
which place? @heyste 

this place?

https://github.com/kubernetes/kubernetes/blob/b3dd01da497f4f9f81ffb2ebf275ff08cbfebaa2/test/e2e/network/service.go#L744-L754

Reply:
leaving it for e2e framework clean is also ok. :)